rust/lance/src/dataset/optimize.rs

```diff
@@ -226,6 +340,13 @@ pub async fn compact_files(
 
     let compaction_plan: CompactionPlan = plan_compaction(dataset, &options).await?;
 
+    if compaction_plan.tasks().is_empty() && options.enable_binary_copy_force {
```

Comment:
Why is this an error instead of a no-op?

Reply:
Actually, it's not needed. This logic has already been removed.